AmplifyPlugins/DataStore/Tests/AWSDataStorePluginTests/Subscribe/ObserveQueryTaskRunnerTests.swift

```diff
@@ -64,6 +72,104 @@ class ObserveQueryTaskRunnerTests: XCTestCase {
         await waitForExpectations([secondSnapshot], timeout: 10)
     }
     
+    ///  Items observed will be returned in a single snapshot on isSynced toggled
+    ///  from false to true. The operation will internally listen to .modelSynced event from
+    ///  the Hub.
+    ///
+    /// - Given:  The operation has started and the first query has completed.
+    /// - When:
+    ///    - Observe items when .modelSynced internally in the Hub
+    /// - Then:
+    ///    - The items observed will be returned in the second snapshot
+    ///
+    func testGenerateSnapshotOnObserveQueryWhenModelSynced() async throws {
+        let firstSnapshot = asyncExpectation(description: "first query snapshots")
+        let secondSnapshot = asyncExpectation(description: "second query snapshots")
+        let dispatchedModelSyncedEvent = AtomicValue(initialValue: false)
+        let taskRunner = ObserveQueryTaskRunner(
+            modelType: Post.self,
+            modelSchema: Post.schema,
+            predicate: nil,
+            sortInput: nil,
+            storageEngine: storageEngine,
+            dataStorePublisher: dataStorePublisher,
+            dataStoreConfiguration: .default,
+            dispatchedModelSyncedEvent: dispatchedModelSyncedEvent,
+            dataStoreStatePublisher: dataStoreStateSubject.eraseToAnyPublisher())
+        let snapshots = taskRunner.sequence
+        Task {
+            var querySnapshots = [DataStoreQuerySnapshot<Post>]()
+            do {
+                for try await querySnapshot in snapshots {
+                    querySnapshots.append(querySnapshot)
+                    if querySnapshots.count == 1 {
+                        XCTAssertEqual(querySnapshot.items.count, 0)
+                        XCTAssertEqual(querySnapshot.isSynced, false)
+                        await firstSnapshot.fulfill()
+                    } else if querySnapshots.count == 2 {
+                        XCTAssertEqual(querySnapshot.items.count, 0)
+                        XCTAssertEqual(querySnapshot.isSynced, true)
+                        await secondSnapshot.fulfill()
+                    }
+                }
+            } catch {
+                XCTFail("Failed with error \(error)")
+            }
+        }
+        
+        await waitForExpectations([firstSnapshot], timeout: 1)
+        
+        dispatchedModelSyncedEvent.set(true)
+        let readyEventPayload = HubPayload(eventName: HubPayload.EventName.DataStore.modelSynced)
```

Comment:
let's name it `let modelSyncedEventPayload` and it should contain `ModelSyncedEvent` as payload with the correct modelName

Reply:
within this test, we can also test for additional events where the modelSyncEvent contains a modelName that does not match, and having an inverted expectation for the 3rd querySnapshot